### PR TITLE
Add binding address to discover_ssdp

### PR DESCRIPTION
**Pull request by glimchb**
In case interface has multiple IP addresses,
discovery will not work correctly
So adding new option to bind to specific address
if provided as an argument

### CODE REVIEW

**Comment by glimchb** [comment]
@mraineri @MoshiBin please review

**Comment by MoshiBin** [comment]
I can confirm that in the case of multiple ipv6 addresses you have to bind to a specific one in order to get responses back to the same interface. See similar behavior here: https://github.com/MoshiBin/ssdpy/blob/master/ssdpy/client.py#L42-L43

File: src/redfish/discovery/discovery.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def sanitize(number, minimum, maximum=None):
     return number
 
 
-def discover_ssdp(port=1900, ttl=2, response_time=3, iface=None, protocol="ipv4"):
+def discover_ssdp(port=1900, ttl=2, response_time=3, iface=None, address=None, protocol="ipv4"):
```
Comment (billdodd) [inline src/redfish/discovery/discovery.py:44]: Suggest adding the new optional param at the end
```suggestion
def discover_ssdp(port=1900, ttl=2, response_time=3, iface=None, protocol="ipv4", address=None):

```

**Comment by MoshiBin** [comment]
Highly suggest documenting this param. Especially noting that it’s crucial when the interface has more than one address. Also, can this be used in place of the iface param?

**Comment by billdodd** [comment]
> Also, can this be used in place of the iface param?

Good point. Looking at the code for the `iface` option, I don't think it is portable - `SO_BINDTODEVICE` is not standard on all OSes. So using the `address` option seems like the better approach. Though we may want to leave `iface` there in case anyone is currently using it successfully, but mark it as deprecated.

File: src/redfish/discovery/discovery.py (hunk 1 above)
Comment (mraineri) [reply src/redfish/discovery/discovery.py]: Fix made

**Comment by mraineri** [comment]
Approved 8/7

**Pull request merged**